File: Assignment5/TestIf.pas
program TestIf;

var
    i, j, k : integer;
    x, y : real;
    p, q : boolean;
    str, txt : string;
    
begin
    i := 1;
    j := 2;
    
    IF i = j THEN x := 3.14
             ELSE x := -5;
     
    IF i <> j THEN y := 3.14
              ELSE y := -5;
    
    writeln('i = ', i, ', j = ', j, ', x = ', x:5:2, ', y = ', y:5:2);
    
    IF i = j THEN BEGIN
        x := -7
    END 
    ELSE BEGIN
        x := 8;
    END;
     
    IF i <> j THEN BEGIN
        y := 14
    END 
    ELSE BEGIN
        y := -2;
    END;
    
    writeln('i = ', i, ', j = ', j, ', x = ', x:5:2, ', y = ', y:5:2);
    
    p := i = j;
    IF p THEN x := 55.55
         ELSE BEGIN
            x := 77.77;
            y := 88.88;
         END;
    
    writeln('i = ', i, ', j = ', j, ', x = ', x:5:2, ', y = ', y:5:2);
    
    q := i <= j;
    k := 10;
    
    if p then i := 33 else if not q then i := 44 else if i = j then i := 55 else i := 6;
    if not p then if q then j := 9 else j := -9;
    if p then if q then k := 11 else k := 12;
    
    writeln('i = ', i, ', j = ', j, ', x = ', x:5:2, ', y = ', y:5:2);
    writeln('k = ', k);
    
    writeln;
    if not p and q and (i <> j) and (x < y) then write('Good-bye');
    if not p then if q then if i <> j then if x < y then writeln(', world!');
end.

File: PscToC++/C++/TestIf.cpp
#include <iostream>
#include <iomanip>
#include <chrono>
#include <string>

using namespace std;
using namespace std::chrono;

int i;
int j;
int k;
double x;
double y;
bool p;
bool q;
string str;
string txt;

int main(int argc, char *args[])
{
    auto _start = steady_clock::now();

    i = 1;
    j = 2;

    if(i == j) {
x = 3.14;
    } else {x = -5;
}

    if(i != j) {
y = 3.14;
    } else {y = -5;
}
    printf("i = %d, j = %d, x = %5.2f, y = %5.2f\n", i, j, x, y);

    if(i == j) {
{
            x = -7;
        }
    } else {{
            x = 8;
        }
}

    if(i != j) {
{
            y = 14;
        }
    } else {{
            y = -2;
        }
}
    printf("i = %d, j = %d, x = %5.2f, y = %5.2f\n", i, j, x, y);
    p = i == j;

    if(p) {
x = 55.55;
    } else {{
            x = 77.77;
            y = 88.88;
        }
}
    printf("i = %d, j = %d, x = %5.2f, y = %5.2f\n", i, j, x, y);
    q = i <= j;
    k = 10;

    if(p) {
i = 33;
    } else {
        if(!q) {
i = 44;
        } else {
            if(i == j) {
i = 55;
            } else {i = 6;
}
}
}

    if(!p) {
        if(q) {
j = 9;
        } else {j = -9;
}
    } else {}

    if(p) {
        if(q) {
k = 11;
        } else {k = 12;
}
    } else {}
    printf("i = %d, j = %d, x = %5.2f, y = %5.2f\n", i, j, x, y);
    printf("k = %d\n", k);
    cout << endl;

    if(!p && q && (i != j) && (x < y)) {
printf("Good-bye");
    } else {}

    if(!p) {
        if(q) {
            if(i != j) {
                if(x < y) {
printf(", world!\n");
                } else {}
            } else {}
        } else {}
    } else {}

    auto _end = steady_clock::now();
    long _elapsed = duration_cast<milliseconds>(_end - _start).count();
    printf("\n[%ld milliseconds execution time.]\n", _elapsed);
}
